UnitsNet.Tests/CustomCode/MassConcentrationTests.cs

```diff
@@ -0,0 +1,159 @@
+        protected override double GramsPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-6;
+        protected override double KilogramsPerCubicCentimeterInOneKilogramPerCubicMeter => 1e-6;
+        protected override double KilogramsPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-9;
+        protected override double TonnesPerCubicCentimeterInOneKilogramPerCubicMeter => 1e-9;
+        protected override double TonnesPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-12;
+
+        protected override double PoundsPerImperialGallonInOneKilogramPerCubicMeter => 1.002241e-2;
+        protected override double KilopoundsPerCubicFootInOneKilogramPerCubicMeter => 6.242796e-5;
+        protected override double KilopoundsPerCubicInchInOneKilogramPerCubicMeter => 3.6127292e-8;
+        protected override double PoundsPerCubicFootInOneKilogramPerCubicMeter => 6.242796e-2;
+        protected override double PoundsPerCubicInchInOneKilogramPerCubicMeter => 3.61272923e-5;
+        protected override double PoundsPerUSGallonInOneKilogramPerCubicMeter => 8.3454045e-3;
+        protected override double SlugsPerCubicFootInOneKilogramPerCubicMeter => 0.00194032;
+        #endregion
+
+        [Theory]
+        [InlineData(2, MassConcentrationUnit.KilogramPerCubicMeter, 3, VolumeUnit.CubicMeter, 6, MassUnit.Kilogram)]
+        public static void MassConcentrationTimesVolumeEqualsMass(
+            double massConcValue, MassConcentrationUnit massConcUnit,
+            double volumeValue, VolumeUnit volumeUnit,
+            double expectedMassValue, MassUnit expectedMassUnit, double tolerance= 1e-5)
+        {
+            var massConcentration = new MassConcentration(massConcValue, massConcUnit);
+            var volume = new Volume(volumeValue, volumeUnit);
+
+            Mass mass = massConcentration * volume;
+            AssertEx.EqualTolerance(mass.As(expectedMassUnit), expectedMassValue, tolerance);
+        }
+
+        [Fact]
+        public static void VolumeTimesMassConcentrationEqualsMass()
+        {
+            Mass mass = Volume.FromCubicMeters(3) * MassConcentration.FromKilogramsPerCubicMeter(2);
+            Assert.Equal(mass, Mass.FromKilograms(6));
+        }
+
+        [Fact]
+        public void ExpectMolarityConvertedToMassConcentrationCorrectly()
+        {
+            var molarity = Molarity.FromMolesPerLiter(1.02698355);
+            var molarMass = MolarMass.FromGramsPerMole(58.443);
+
+            MassConcentration concentration = molarity.ToMassConcentration(molarMass);     // molarity * molarMass
+            AssertEx.EqualTolerance(60.02, concentration.KilogramsPerCubicMeter, KilogramsPerCubicMeterTolerance);
+        }
+
+        [Theory]
+        [InlineData(60.02, MassConcentrationUnit.KilogramPerCubicMeter, 58.443, MolarMassUnit.GramPerMole, 1026.98355, MolarityUnit.MolesPerCubicMeter)]
+        public void ExpectMassConcentrationConvertedToMolarityCorrectly(
```

Comment:
Empty line between Act-code and Assert-code would make it easier to read, per above discussion on AAA-pattern. 
Also consider renaming method to describe what method is being tested, what are the inputs and what is the expected outcome (if feasible). Same for other tests.